File: ios/Library/Themes/Default/Auth/SignUpScreenlet/SignUpView_default.swift
/**
* Copyright (c) 2000-present Liferay, Inc. All rights reserved.
*
* This library is free software; you can redistribute it and/or modify it under
* the terms of the GNU Lesser General Public License as published by the Free
* Software Foundation; either version 2.1 of the License, or (at your option)
* any later version.
*
* This library is distributed in the hope that it will be useful, but WITHOUT
* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
* FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
* details.
*/
import UIKit


public class SignUpView_default: BaseScreenletView, SignUpData {

	@IBOutlet internal var emailAddressField: UITextField?
	@IBOutlet internal var passwordField: UITextField?
	@IBOutlet internal var firstNameField: UITextField?
	@IBOutlet internal var lastNameField: UITextField?
	@IBOutlet internal var signUpButton: UIButton?
	@IBOutlet internal var emailAddressBackground: UIImageView?
	@IBOutlet internal var passwordBackground: UIImageView?
	@IBOutlet internal var firstNameBackground: UIImageView?
	@IBOutlet internal var lastNameBackground: UIImageView?

	@IBOutlet internal var scrollView: UIScrollView?


	//MARK: BaseScreenletView

	override internal func onStartOperation() {
		signUpButton!.enabled = false
	}

	override internal func onFinishOperation() {
		signUpButton!.enabled = true
	}


	//MARK: BaseScreenletView

	override internal func onCreated() {
		super.onCreated()

		setButtonDefaultStyle(signUpButton)

		scrollView?.contentSize = scrollView!.frame.size

		BaseScreenlet.setHUDCustomColor(DefaultThemeBasicBlue)
	}

	override func onSetTranslations() {
		firstNameField?.placeholder = LocalizedString("default", "first-name-placeholder", self)
		lastNameField?.placeholder = LocalizedString("default", "last-name-placeholder", self)
		emailAddressField?.placeholder = LocalizedString("default", "auth-method-email", self)
		passwordField?.placeholder = LocalizedString("default", "password-placeholder", self)

		signUpButton?.replaceAttributedTitle(
				LocalizedString("default", "sign-up-button", self),
				forState: .Normal)
	}

	//MARK: SignUpData

	public var emailAddress: String? {
		get {
			return nullIfEmpty(emailAddressField!.text)
		}
		set {
			emailAddressField!.text = newValue
		}
	}

	public var password: String? {
		get {
			return nullIfEmpty(passwordField!.text)
		}
		set {
			passwordField!.text = newValue
		}
	}

	public var firstName: String? {
		get {
			return nullIfEmpty(firstNameField!.text)
		}
		set {
			firstNameField!.text = newValue
		}
	}

	public var lastName: String? {
		get {
			return nullIfEmpty(lastNameField!.text)
		}
		set {
			lastNameField!.text = newValue
		}
	}

	public var companyId: Int64 = 0

	// The following properties are not supported in this theme but
	// may be supported in a child theme

	public var screenName: String? {
		get { return nil }
		set {}
	}

	public var middleName: String? {
		get { return nil }
		set {}
	}

	public var jobTitle: String? {
		get { return nil }
		set {}
	}


	//MARK: UITextFieldDelegate

	internal func textFieldDidBeginEditing(textField: UITextField!) {
		emailAddressBackground?.highlighted = (textField == emailAddressField)
		passwordBackground?.highlighted = (textField == passwordField)
		firstNameBackground?.highlighted = (textField == firstNameField)
		lastNameBackground?.highlighted = (textField == lastNameField)
	}

}

File: xamarin/Framework/LiferayScreens.iOS/Themes/Default/SignUpView_default.cs
using Foundation;
using ObjCRuntime;
using CoreGraphics;
using System;
using UIKit;

namespace LiferayScreens
{
    // @interface SignUpView_default : BaseScreenletView <SignUpViewModel>
    [BaseType(typeof(BaseScreenletView))]
    interface SignUpView_default : ISignUpViewModel
    {
        // @property (nonatomic, strong) UITextField * _Nullable emailAddressField __attribute__((iboutlet));
        [NullAllowed, Export("emailAddressField", ArgumentSemantic.Strong)]
        UITextField EmailAddressField { get; set; }

        // @property (nonatomic, strong) UITextField * _Nullable passwordField __attribute__((iboutlet));
        [NullAllowed, Export("passwordField", ArgumentSemantic.Strong)]
        UITextField PasswordField { get; set; }

        // @property (nonatomic, strong) UITextField * _Nullable firstNameField __attribute__((iboutlet));
        [NullAllowed, Export("firstNameField", ArgumentSemantic.Strong)]
        UITextField FirstNameField { get; set; }

        // @property (nonatomic, strong) UITextField * _Nullable lastNameField __attribute__((iboutlet));
        [NullAllowed, Export("lastNameField", ArgumentSemantic.Strong)]
        UITextField LastNameField { get; set; }

        // @property (nonatomic, strong) UIButton * _Nullable signUpButton __attribute__((iboutlet));
        [NullAllowed, Export("signUpButton", ArgumentSemantic.Strong)]
        UIButton SignUpButton { get; set; }

        // @property (nonatomic, strong) UIScrollView * _Nullable scrollView __attribute__((iboutlet));
        [NullAllowed, Export("scrollView", ArgumentSemantic.Strong)]
        UIScrollView ScrollView { get; set; }

        // -(void)onStartInteraction;
        [Export("onStartInteraction")]
        void OnStartInteraction();

        // -(void)onFinishInteraction:(id _Nullable)result error:(NSError * _Nullable)error;
        [Export("onFinishInteraction:error:")]
        void OnFinishInteraction([NullAllowed] NSObject result, [NullAllowed] NSError error);

        // -(void)onCreated;
        [Export("onCreated")]
        void OnCreated();

        // -(void)onSetTranslations;
        [Export("onSetTranslations")]
        void OnSetTranslations();

        // -(id<ProgressPresenter> _Nonnull)createProgressPresenter __attribute__((warn_unused_result));
        //[Export("createProgressPresenter")]
        //[Verify(MethodToProperty)]
        //ProgressPresenter CreateProgressPresenter { get; }

        // @property (copy, nonatomic) NSString * _Nullable emailAddress;
        [NullAllowed, Export("emailAddress")]
        string EmailAddress { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable password;
        [NullAllowed, Export("password")]
        string Password { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable firstName;
        [NullAllowed, Export("firstName")]
        string FirstName { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable lastName;
        [NullAllowed, Export("lastName")]
        string LastName { get; set; }

        // @property (nonatomic) BOOL editCurrentUser;
        [Export("editCurrentUser")]
        bool EditCurrentUser { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable screenName;
        [NullAllowed, Export("screenName")]
        string ScreenName { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable middleName;
        [NullAllowed, Export("middleName")]
        string MiddleName { get; set; }

        // @property (copy, nonatomic) NSString * _Nullable jobTitle;
        [NullAllowed, Export("jobTitle")]
        string JobTitle { get; set; }

        // -(instancetype _Nonnull)initWithFrame:(CGRect)frame __attribute__((objc_designated_initializer));
        [Export("initWithFrame:")]
        [DesignatedInitializer]
        IntPtr Constructor(CGRect frame);
    }
}
